src/ORM/Table.php

```diff
@@ -695,39 +695,34 @@ public function setDisplayField($field)
     /**
      * Returns the display field.
      *
-     * @return array<string>|string|null
+     * @return array<string>|string
      */
     public function getDisplayField()
     {
-        if ($this->_displayField === null) {
-            $displayFieldFound = false;
-            $schema = $this->getSchema();
-            $this->_displayField = $this->getPrimaryKey();
-            $expectedFields = ['title', 'name', 'label'];
-            foreach ($expectedFields as $field) {
-                if ($schema->hasColumn($field)) {
-                    $this->_displayField = $field;
-                    $displayFieldFound = true;
-                    break;
-                }
+        if ($this->_displayField !== null) {
+            return $this->_displayField;
+        }
+
+        $schema = $this->getSchema();
+        foreach (['title', 'name', 'label'] as $field) {
+            if ($schema->hasColumn($field)) {
+                return $this->_displayField = $field;
             }
-            if (!$displayFieldFound) {
-                foreach ($schema->columns() as $column) {
-                    $columnSchema = $schema->getColumn($column);
-                    if (
-                        $columnSchema &&
-                        $columnSchema['null'] !== true &&
-                        $columnSchema['type'] === 'string' &&
-                        !preg_match('/pass|token|secret/', $column)
-                    ) {
-                        $this->_displayField = $column;
-                        break;
-                    }
-                }
+        }
+
+        foreach ($schema->columns() as $column) {
+            $columnSchema = $schema->getColumn($column);
+            if (
+                $columnSchema &&
+                $columnSchema['null'] !== true &&
+                $columnSchema['type'] === 'string' &&
+                !preg_match('/pass|token|secret/', $column)
```

Comment:
```suggestion
                !preg_match('/pass|token|secret/i', $column)
```
This should probably be case-insensitive.